Rewrap all comments to 100 line width

kianenigma opened the pull request:
Closes https://github.com/paritytech/substrate/issues/6222

This makes the code much more readable for anyone who has visual rules in their editor, and just much more appealing over all. 

I have to go through them once and revert maybe the links and bash commands in comments.

kianenigma commented on bin/node/cli/src/chain_spec.rs:
```diff
@@ -78,9 +78,10 @@ fn session_keys(
-	// for i in 1 2 3 4 ; do for j in session; do subkey --ed25519 inspect "$secret"//fir//$j//$i; done; done
+	// for i in 1 2 3 4 ; do for j in stash controller; do subkey inspect "$secret"/fir/$j/$i; done;
+	// done and
+	// for i in 1 2 3 4 ; do for j in session; do subkey --ed25519 inspect "$secret"//fir//$j//$i;
```
needs revert

kianenigma commented on client/authority-discovery/src/worker.rs:
```diff
@@ -91,7 +91,8 @@ pub enum Role {
 ///
-/// When constructed with either [`Role::PublishAndDiscover`] or [`Role::Discover`] a [`Worker`] will
+/// When constructed with either [`Role::PublishAndDiscover`] or [`Role::Discover`] a [`Worker`]
+/// will
```
?????

kianenigma commented on client/service/src/client/call_executor.rs:
```diff
@@ -253,7 +253,8 @@ where
 				.set_parent_hash(at_hash);
 				// TODO: https://github.com/paritytech/substrate/issues/4455
-				// .with_storage_transaction_cache(storage_transaction_cache.as_mut().map(|c| &mut **c))
+				// .with_storage_transaction_cache(storage_transaction_cache.as_mut().map(|c| &mut
```
a bit strange now, but this comment is kinda invalid anyway. The example code should be in the issue.

tomaka commented on rustfmt.toml:
```diff
@@ -7,6 +7,9 @@ imports_granularity = "Crate"
 # Consistency
 newline_style = "Unix"
+# Format comments
+comment_width = 100
+wrap_comments = true
```
These options are unstable. I don't know exactly what happens, but I expect stable `rustfmt` to not be too happy.

athei replied:
We are already using unstable format options without this.

kianenigma replied:
indeed.

gilescope replied:
It's pretty easy to call the nightly rustfmt from stable. Potentially we can push to stabilise the features we're using - quite often things are unstable until someone drops a PR to propose to stabilise them.

athei commented:
I did not review all files (obviously) but from what I have seen it did improve the visual appearance.

gilescope commented on client/service/test/src/client/mod.rs:
```diff
@@ -1826,8 +1826,8 @@ fn imports_blocks_with_changes_tries_config_change() {
 	// block 27 changes the key
-	// block 28 is the L1 digest (NOT SKEWED!!!) that covers changes AND changes configuration to 3^1
-	// ===================================================================
+	// block 28 is the L1 digest (NOT SKEWED!!!) that covers changes AND changes configuration to
+	// 3^1 ===================================================================
```
this needs shaping

gilescope commented on client/state-db/src/noncanonical.rs:
```diff
@@ -870,13 +874,10 @@ mod tests {
-		//     \ 2_2
+		// - 1 - 1_1 - 1_1_1 \ 1_2 - 1_2_1 \ 1_2_2 \ 1_2_3
+		//
+		// - 2 - 2_1 - 2_1_1 \ 2_2
```
oh dear

kianenigma replied:
did `#[rustfmt::skip]` for this one.

expenses replied:
Could you re-do it please? Seems to be broken.

gilescope commented on client/state-db/src/noncanonical.rs:
```diff
@@ -1084,8 +1084,8 @@ mod tests {
-		// - 0 - 1_1 - 2_1
-		//     \ 1_2
+		//
+		// - 0 - 1_1 - 2_1 \ 1_2
```
and here

gilescope commented on frame/aura/src/lib.rs:
```diff
@@ -89,17 +90,8 @@ pub mod pallet {
-
-				// TODO [#3398] Generate offence report for all authorities that skipped their slots.
+				// TODO [#3398] Generate offence report for all authorities that skipped their
+				// slots.
```
We seemed to loose some code here?

kianenigma replied:
not sure what do you mean?

gilescope replied:
I'm seeing these lines removed in the diff for frame/aura/src/lib.rs:
```
if let Some(n_authorities) = <Authorities<T>>::decode_len() {
					let authority_index = *new_slot % n_authorities as u64;
				// TODO [#3398] Generate offence report for all authorities that skipped their slots.
```
I think the branch has managed to drop some of this commit: https://github.com/paritytech/substrate/commit/471db4f2b6ef9e9871fd1e9af2a56050efb3beb6

gilescope replied:
I think the github ui is a tad dodgy here. I had to click into the file and expand it a bit before it showed me the correct diff. (First time I've see what looks like a github ui bug.)

kianenigma replied:
yeah same, it is still showing a wrong diff to me, I'll do a local compare before merge

gilescope commented on frame/democracy/src/lib.rs:
```diff
@@ -1632,11 +1637,12 @@ impl<T: Config> Pallet<T> {
+			                                                            * current public
+			                                                            * proposals have an
+			                                                            * amount locked */
+		) {
```
needs a revisit.

gilescope commented on frame/example/src/lib.rs:
```diff
@@ -37,13 +37,13 @@
+//! <code>INHERENT_IDENTIFIER</code>, <code>InherentType</code>,         <code>u64</code>)</li>
+//!     <li>Usage scenarios - should be simple doctests. The compiler should ensure they stay
+//! valid.</li>     <li>Extended tutorials - should be moved to external files and refer to.</li>
 //!     <!-- Original author of paragraph: @AmarRSingh -->
```
how does this look in rust doc?

kianenigma replied:
I just aignored this doc block, I don't know why we have HTML stuffed in there.

expenses replied:
Hmmm, this isn't great

gilescope commented on frame/example/src/lib.rs:
```diff
@@ -522,17 +527,17 @@ pub mod pallet {
 
-			// Print out log or debug message in the console via log::{error, warn, info, debug, trace},
-			// accepting format strings similar to `println!`.
+			// Print out log or debug message in the console via log::{error, warn, info, debug,
+			// trace}, accepting format strings similar to `println!`.
```
```suggestion
			// Print out log or debug message in the console via 
			// `log::{error, warn, info, debug, trace}`
			// accepting format strings similar to `println!`.
```

gilescope commented on frame/support/src/storage/weak_bounded_vec.rs:
```diff
@@ -270,8 +270,8 @@ impl<T, S> codec::DecodeLength for WeakBoundedVec<T, S> {
 // NOTE: we could also implement this as:
-// impl<T: Value, S1: Get<u32>, S2: Get<u32>> PartialEq<WeakBoundedVec<T, S2>> for WeakBoundedVec<T, S1>
-// to allow comparison of bounded vectors with different bounds.
+// impl<T: Value, S1: Get<u32>, S2: Get<u32>> PartialEq<WeakBoundedVec<T, S2>> for WeakBoundedVec<T,
+// S1> to allow comparison of bounded vectors with different bounds.
```
not ideal

gilescope commented on primitives/runtime-interface/src/pass_by.rs:
```diff
@@ -86,19 +87,19 @@ pub use sp_runtime_interface_proc_macro::PassByInner;
+/// 	Okay,
+/// 	NotOkay,
+/// 	/* This will not work with the derive.
+/// 	 *Why(u32), */
```
Interesting...

gilescope commented on utils/fork-tree/src/lib.rs:
```diff
@@ -912,14 +912,14 @@ mod test {
+		//           - L - M \
 		//               - O
 		//  \
 		//   — J - K
```
So close. It _almost_ didn't mess this up.

gilescope commented on utils/wasm-builder/src/builder.rs:
```diff
@@ -219,8 +219,8 @@ fn generate_rerun_if_changed_instructions() {
-/// `wasm_binary_name` - The optional wasm binary name that is extended with `.compact.compressed.wasm`.
-///                      If `None`, the project name will be used.
+/// `wasm_binary_name` - The optional wasm binary name that is extended with
+/// `.compact.compressed.wasm`.                      If `None`, the project name will be used.
```
```suggestion
/// `.compact.compressed.wasm`. If `None`, the project name will be used.
```

kianenigma commented on client/db/src/cache/list_cache.rs:
```diff
@@ -1015,8 +1017,8 @@ mod tests {
-		// when block is later than last finalized block AND there are no forks AND finalized value is Some
-		// ---> [100] --- 200
+		// when block is later than last finalized block AND there are no forks AND finalized value
+		// is Some ---> [100] --- 200
```
this one looks fine to me.

kianenigma commented on client/db/src/lib.rs:
```diff
@@ -3395,7 +3396,8 @@ pub(crate) mod tests {
 
-		// Insert 1 as best again. This should fail because canonicalization_delay == 3 and best == 5
+		// Insert 1 as best again. This should fail because canonicalization_delay == 3 and best ==
+		// 5
```
these can and should be fixed by a '' wrapping.

kianenigma commented:
@gilescope I belive I have addressed all your issues, except a handful of ones that I didn't resolve. 

I am still strongly for this change. Most of the things that we had to fix manually are either code snippets, which can be `#[rustfmt::skipp]`ed, or due to using non-standard markdown.

expenses commented on frame/bounties/src/tests.rs:
```diff
@@ -774,7 +774,8 @@ fn cancel_and_refund() {
 		assert_ok!(Bounties::close_bounty(Origin::root(), 0));
 
-		assert_eq!(Treasury::pot(), 85); // - 25 + 10
+		assert_eq!(Treasury::pot(), 85); // 
+		                         // - 25 + 10
```
This is a bit odd.

expenses commented on frame/democracy/src/lib.rs:
```diff
@@ -1620,11 +1625,12 @@ impl<T: Config> Pallet<T> {
 	/// Table the waiting public proposal with the highest backing for a vote.
 	fn launch_public(now: T::BlockNumber) -> DispatchResult {
 		let mut public_props = Self::public_props();
-		if let Some((winner_index, _)) = public_props.iter()
-			.enumerate()
-			.max_by_key(|x| Self::backing_for((x.1).0).unwrap_or_else(Zero::zero)
-				/* ^^ defensive only: All current public proposals have an amount locked*/)
-		{
+		if let Some((winner_index, _)) = public_props.iter().enumerate().max_by_key(
+			|x| Self::backing_for((x.1).0).unwrap_or_else(Zero::zero), /* ^^ defensive only: All
+			                                                            * current public
+			                                                            * proposals have an
+			                                                            * amount locked */
+		) {
```
Also a bit odd

expenses commented on frame/example/src/lib.rs:
```diff
@@ -130,14 +132,15 @@
-//! // Reference documentation of aspects such as `storageItems` and `dispatchable` functions should only be
-//! // included in the <https://docs.rs> Rustdocs for Substrate and not repeated in the README file.
+//! // Reference documentation of aspects such as `storageItems` and `dispatchable` functions should
+//! only be // included in the <https://docs.rs> Rustdocs for Substrate and not repeated in the README file.
```
What's with the `//! //`s here?

kianenigma replied:
it is because the previous doc was doing `//! //`, which is quite confusing to rustfmt.

expenses commented on frame/identity/src/lib.rs:
```diff
@@ -451,7 +451,8 @@ pub mod pallet {
 			Ok(Some(
 				T::WeightInfo::set_subs_old(old_ids.len() as u32) // P: Real number of old accounts removed.
-					.saturating_add(T::WeightInfo::set_subs_new(new_subs as u32)), /* S: New subs added. */
+					.saturating_add(T::WeightInfo::set_subs_new(new_subs as u32)), /* S: New subs
+				                                                                 * added. */
```
Hmm

expenses commented on frame/support/src/lib.rs:
````diff
@@ -2271,7 +2275,8 @@ pub mod pallet_prelude {
 /// 	impl<T: Config> Hooks for Pallet<T> {
 /// 	}
 /// 	```
-/// 	and write inside on_initialize/on_finalize/on_runtime_upgrade/offchain_worker/integrity_test
+/// 	and write inside
+/// on_initialize/on_finalize/on_runtime_upgrade/offchain_worker/integrity_test
````
Think we have too many spaces here?

expenses commented on primitives/npos-elections/src/reduce.rs:
```diff
@@ -506,8 +506,8 @@ fn reduce_all<A: IdentifierT>(assignments: &mut Vec<StakedAssignment<A>>) -> u32
 										if next_value.is_zero() {
-											// if the removed edge is from the current assignment, dis_index
-											// should NOT be increased.
+											// if the removed edge is from the current assignment,
+											// dis_index should NOT be increased.
```
Nitpicking but should this be `dis_index`?

kianenigma replied:
honestly I am not sure what that is :D I changed it to `index`.

expenses reviewed (changes requested):
99% of these changes look great :D ! But the remaining 1% do not. I think you'll need to manually change a few comments and then re-format.

expenses reviewed (approved):
I don't really to review the whole thing again, but I think that if you ran `cargo +nightly fmt` again it should be good to go.

kianenigma commented:
bot merge

kianenigma commented:
bot merge
